astropy/units/core.py

```diff
@@ -1733,6 +1742,56 @@ def is_unity(self):
         return False
 
 
+class ArbitraryUnit(IrreducibleUnit):
+    """
+    A unit that has a completely arbitrary dimension.  This useful for, e.g.,
+    Quantities that are all 0 or infinity.  There can be only one such unit.
+
+    Parameters
+    ----------
+    namespace : dict, optional
+        When provided, inject the unit, and all of its aliases, in the
+        given namespace dictionary.  If a unit by the same name is
+        already in the namespace, a ValueError is raised.
+    """
+    _arbitrary_unit = None
+
+    def __new__(cls, namespace=None):
+        if cls._arbitrary_unit is None:
+            return super(ArbitraryUnit, cls).__new__(cls)
+
+        if namespace is not None:
+            cls._arbitrary_unit._inject(namespace)
+
+        return cls._arbitrary_unit
+
+    def __init__(self, namespace=None):
+        if self.__class__._arbitrary_unit is None:
+            super(ArbitraryUnit, self).__init__(
+                st=['arbitrary', 'arbitrary_unit'], doc='Arbitrary unit',
+                format=None, namespace=namespace)
+            self.__class__._arbitrary_unit = self
+
+    def decompose(self, bases=set()):
+        if len(bases) and not self in bases:
+            return bases[0]
+        else:
+            return self
```

Comment:
Hmm, clearly a part I didn't write a test case for...  The idea is indeed that if one gives a set to decompose into, it can be any of those, so one just picks one. But maybe it is better simply to raise `ValueError` for this case?